python/tskit/trees.py

```diff
@@ -5085,17 +5085,18 @@ def individual_populations(self):
         has nodes with inconsistent non-NULL populations.
         """
         if self._individual_populations is None:
-            pops = np.repeat(tskit.NULL, self.num_individuals)
+            pops = np.full(self.num_individuals, tskit.NULL, dtype=np.int32)
+            seen = np.full(self.num_individuals, False, dtype=bool)
```

Comment:
You don't need the full ``seen`` array - you can look at the local values for a given individual. In C-like code you'd have:
```
for ind in self.individuals():
     if len(ind.nodes) > 0: 
          ind_pop = -2
          for u in ind.nodes:
               if ind_pop == -2:
                  ind_pop = tables.nodes.population[u]
               elif ind_pop != tables.nodes.population[u]:
                   raise ValueError(..)
           population[ind.id] = ind_pop
```

Reply:
Sorry, I'm confused - in C we won't have ready access to the individual's nodes when iterating over individuals, right?  Oh! I see, we *do*! Sorry, I was looking at the `stable` documentation, which I see hasn't updated to C 1.0 yet. Yes, that is a lot easier.